deepchem/molnet/load_function/bbbc_datasets.py

```diff
@@ -146,3 +148,176 @@ def load_bbbc002(
+    fluorescent cell population images. There are 300 cells in each sample, each an image
+    of size 950x950. Ground truth labels contain cell counts and segmentation masks for
+    this dataset. Full details about this dataset are present at
+    https://data.broadinstitute.org/bbbc/BBBC004/.
+
+    Parameters
+    ----------
+    overlap_probability: float from list {0.0, 0.15, 0.3, 0.45, 0.6}
+        the overlap probability of the synthetic cells in the images
+    load_segmentation_mask: bool
+        if True, the dataset will contain segmentation masks as labels. Otherwise,
+        the dataset will contain cell counts as labels.
+    splitter: Splitter or str
+        the splitter to use for splitting the data into training, validation, and
+        test sets.  Alternatively you can pass one of the names from
+        dc.molnet.splitters as a shortcut.  If this is None, all the data
+        will be included in a single dataset.
+    transformers: list of TransformerGenerators or strings
+        the Transformers to apply to the data.  Each one is specified by a
+        TransformerGenerator or, as a shortcut, one of the names from
+        dc.molnet.transformers.
+    reload: bool
+        if True, the first call for a particular featurizer and splitter will cache
+        the datasets to disk, and subsequent calls will reload the cached datasets.
+    data_dir: str
+        a directory to save the raw data in
+    save_dir: str
+        a directory to save the dataset in
+
+    Examples
+    --------
+    Importing necessary modules
+
+    >>> import deepchem as dc
+    >>> import numpy as np
+
+    We can load the BBBC004 dataset with 2 types of labels: segmentation masks and
+    cell counts. We will first load the dataset with cell counts as labels.
+
+    >>> loader = dc.molnet.load_bbbc004(overlap_probability=0.0, load_segmentation_mask=False)
+    >>> tasks, dataset, transformers = loader
+    >>> train, val, test = dataset
+
+    We now have a dataset with 20 samples, each with 300 cells. The images are of
+    size 950x950. The labels are cell counts. We can verify this as follows:
+
+    >>> train.X.shape
+    (16, 950, 950)
+    >>> train.y.shape
+    (16,)
+
+    We will now load the dataset with segmentation masks as labels.
+
+    >>> loader = dc.molnet.load_bbbc004(overlap_probability=0.0, load_segmentation_mask=True)
+    >>> tasks, dataset, transformers = loader
+    >>> train, val, test = dataset
+
+    We now have a dataset with 20 samples, each with 300 cells. The images are of
+    size 950x950. The labels are segmentation masks. We can verify this as follows:
+
+    >>> train.X.shape
+    (16, 950, 950)
+    >>> train.y.shape
+    (16, 950, 950, 3)
+    """
+    featurizer = dc.feat.UserDefinedFeaturizer([])  # Not actually used
+    if load_segmentation_mask:
+        loader = _BBBC004_Segmentation_Loader(overlap_probability, featurizer,
```

Comment:
Explicitly annotate `loader: _MolnetLoader` since it's getting confused

Reply:
`loader : _MolnetLoader = None` before the if/else

Reply:
Made the edit